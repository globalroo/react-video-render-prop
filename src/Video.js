import React, { Component } from 'react';

class Video extends Component {
	constructor(props) {
		super(props);
		this.state = { currentTime: 0 };
		this.updatePlayerState = this.updatePlayerState.bind(this);
	}

	updatePlayerState() {
		this.setState({
			currentTime: `${this._video.currentTime.toFixed(2)}`,
		});
	}

	componentDidMount() {
		this._video.addEventListener("timeupdate", this.updatePlayerState);
		this._video.src = "https://media.w3.org/2010/05/sintel/trailer.mp4";
	}

	render() {
		return (
			<div>
				<video controls ref={(ref) => this._video = ref} />
				{this.props.renderProp(this.state)}
			</div>
		);
	}
}

export default Video;
