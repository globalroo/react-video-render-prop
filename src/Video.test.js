import React from 'react';
import Video from './Video';
import * as renderer from "react-test-renderer";

const renderPropComponent = (state) => (
	<div> mockComponent {state.currentTime} </div>
);

function handleDOMMocks(element) {
	if (element.type === 'video') {
	  return {
		addEventListener: jest.fn()
	  };
	}
	return null;
  }

describe("Video component tests", () => {
	test("Video component renders as expected", () => {
		const options = {
			createNodeMock: handleDOMMocks
		};
		const component = renderer.create(
			<Video renderProp={renderPropComponent}/>,
			options
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test("updatePlayerState returns the expected state", () => {
		const spy = jest.fn();
		const component = new Video({ renderProp: renderPropComponent });
		component.setState = spy;
		component._video = {
			currentTime: 23.123456
		}
		component.updatePlayerState();
		expect(spy).toHaveBeenCalledWith({"currentTime": "23.12"});
	});
});
