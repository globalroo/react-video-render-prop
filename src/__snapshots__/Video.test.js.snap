// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video component tests Video component renders as expected 1`] = `
<div>
  <video
    controls={true}
  />
  <div>
     mockComponent 
    0
     
  </div>
</div>
`;
